README.md

```diff
@@ -24,12 +24,34 @@ ChatGPT uses your private conversations history to train its models. [OpenAI dis
 
 # Table of Contents
 
+- [Configuration](#configuration)
 - [Deploy the app on Render](#deploy-the-app-on-render)
 - [Deploy the app on Fly.io](#deploy-the-app-on-fly)
 - [Contribute as a developer](#contribute-as-a-developer)
 - [Understanding the Docker configuration](#understanding-the-docker-configuration)
 - [Changelog](#changelog)
 
+# Configuration
+
+Following features can be configured with environment variables:
+
+- `REGISTRATON_FEATURE` - Set to `false` to disable registration.
+  Default is `true`.
+- `AUTHENTICATION_APPROACH` - Set to `google` to use Google OAuth for authentication, `http_header` to use HTTP header authentication, or `password` to use password authentication.
+  Default is `password`.
+- `AUTHENTICATION_HTTP_HEADER_EMAIL` - Name of the HTTP request header, which can be used to derive an email address of a new user.
+  Required if `AUTHENTICATION_APPROACH` is set to `http_header`.
+  Default is `Tailscale-User-Login`.
+- `AUTHENTICATION_HTTP_HEADER_NAME` - Name of the HTTP request header, which can be used to derive the name of a new user.
+  Required if `AUTHENTICATION_APPROACH` is set to `http_header`.
+  Default is `Tailscale-User-Name`.
+- `AUTHENTICATION_HTTP_HEADER_UID` - Name of the HTTP request header, which will be used as unique identifier of the user.
+  Required if `AUTHENTICATION_APPROACH` is set to `http_header`.
+  Default is `Tailscale-User-Login`.
+- `GOOGLE_CLIENT_ID` - Google OAuth client ID.
+  Required if `AUTHENTICATION_APPROACH` is set to `google`.
+- `GOOGLE_CLIENT_SECRET` - Google OAuth client secret.
+  Required if `AUTHENTICATION_APPROACH` is set to `google`.
```

Comment:
If you want to enable google auth with just a normal gmail account (not a custom google org) do you somehow get a google_client_id and secret? Or are those things only required when you’re doing a custom google org/domain?

I’ve never actually implemented google registration before.

Reply:
To enable Google SSO, you will need (a) a Google Cloud account, (b) configure a "consent screen" (c) register your application. Then you get the client id and the client secret.

Reply:
I will add this to the README later

Reply:
Done